src/seedu/addressbook/commands/EditCommand.java

```diff
@@ -0,0 +1,69 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.data.person.UniquePersonList.DuplicatePersonException;
+import seedu.addressbook.data.person.UniquePersonList.PersonNotFoundException;
+
+/**
+ * Edits details of the person identified using the last displayed index.
+ */
+public class EditCommand extends Command {
+
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits some details of the person "
+            + "identified by the index number in the last shown person listing.\n"
+            + "Parameters: INDEX [[p]p/PHONE] [[p]e/EMAIL] [[p]a/ADDRESS]\n"
+            + "Example: " + COMMAND_WORD + " 1 p/23456789 a/311, Clementi Ave 2, #01-01";
+
+    public static final String MESSAGE_EDIT_PERSON_SUCCESS = "Edited Person: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book";
+
+    private Phone phone;
+    private Email email;
+    private Address address;
+
+    /**
+     * Convenience constructor using raw values.
+     *
+     * @throws IllegalValueException if any of the raw values are invalid
+     */
+    public EditCommand (int targetVisibleIndex,
+                        String phone, boolean isPhonePrivate,
+                        String email, boolean isEmailPrivate,
+                        String address, boolean isAddressPrivate) throws IllegalValueException {
+        super(targetVisibleIndex);
+        this.phone = (phone == null) ? null : new Phone(phone, isPhonePrivate);
+        this.email = (email == null) ? null : new Email(email, isEmailPrivate);
+        this.address = (address == null) ? null : new Address(address, isAddressPrivate);
+    }
+
+    @Override
+    public CommandResult execute() {
```

Comment:
Perhaps a comment here would be appropriate to describe what this overridden `execute()` method does.